src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -892,19 +893,30 @@ class ResourceService(
     val addEventSet = if (addEvent.changeDetails.isEmpty) Set.empty else Set(addEvent)
     addEventSet ++ removeEventSet
   }
-
   private def groupFlat(dbResult: Seq[FilterResourcesResult]): FilteredResourcesFlat = {
     val groupedFilteredResource = dbResult
       .groupBy(_.resourceId)
       .map { tuple =>
         val (k, v) = tuple
+        type Accumulated = (Set[FilteredResourceFlatPolicy], Set[ResourceRoleName], Set[ResourceAction], Map[WorkbenchGroupName, Boolean])
+        val base = (Set.empty[FilteredResourceFlatPolicy], Set.empty[ResourceRoleName], Set.empty[ResourceAction], Map.empty[WorkbenchGroupName, Boolean])
+        val grouped = v.foldLeft(base)((acc: Accumulated, r: FilterResourcesResult) =>
+          (
+            acc._1 ++ r.policy.map(p => FilteredResourceFlatPolicy(p, r.isPublic, r.inherited)),
+            acc._2 ++ r.role,
+            acc._3 ++ r.action,
+            acc._4 ++ r.authDomain.map(_ -> r.inAuthDomain)
+          )
+        )
+
         FilteredResourceFlat(
           resourceId = k,
-          resourceType = v.map(_.resourceTypeName).head,
-          policies = v.flatMap(_.policy).toSet,
-          roles = v.flatMap(_.role).toSet,
-          actions = v.flatMap(_.action).toSet,
-          isPublic = v.map(_.isPublic).head
+          resourceType = v.head.resourceTypeName,
+          policies = grouped._1,
+          roles = grouped._2,
+          actions = grouped._3,
+          authDomainGroups = grouped._4.keySet,
+          missingAuthDomainGroups = grouped._4.filter(!_._2).keySet
```

Comment:
This line is a little tricky might be nice to leave a comment explaining what !_._2 is